--- ./libvpx_vp8_encoder.cc	2025-05-19 15:51:58
+++ ./libvpx_vp8_encoder2.cc	2025-05-19 15:56:05
@@ -214,12 +214,14 @@
   return left == right;
 }
 
-void SetRawImagePlanes(vpx_image_t* raw_image, VideoFrameBuffer* buffer) {
+bool SetRawImagePlanes(vpx_image_t* raw_image, VideoFrameBuffer* buffer) {
   switch (buffer->type()) {
     case VideoFrameBuffer::Type::kI420:
     case VideoFrameBuffer::Type::kI420A: {
       const I420BufferInterface* i420_buffer = buffer->GetI420();
-      RTC_DCHECK(i420_buffer);
+      if (!i420_buffer) {
+        return false;
+      }
       raw_image->planes[VPX_PLANE_Y] =
           const_cast<uint8_t*>(i420_buffer->DataY());
       raw_image->planes[VPX_PLANE_U] =
@@ -233,7 +235,9 @@
     }
     case VideoFrameBuffer::Type::kNV12: {
       const NV12BufferInterface* nv12_buffer = buffer->GetNV12();
-      RTC_DCHECK(nv12_buffer);
+      if (!nv12_buffer) {
+        return false;
+      }
       raw_image->planes[VPX_PLANE_Y] =
           const_cast<uint8_t*>(nv12_buffer->DataY());
       raw_image->planes[VPX_PLANE_U] =
@@ -247,6 +251,7 @@
     default:
       RTC_DCHECK_NOTREACHED();
   }
+  return true;
 }
 
 // Helper class used to temporarily change the frame drop threshold for an
@@ -1496,7 +1501,9 @@
   // Prepare `raw_images_` from `mapped_buffer` and, if simulcast, scaled
   // versions of `buffer`.
   std::vector<scoped_refptr<VideoFrameBuffer>> prepared_buffers;
-  SetRawImagePlanes(&raw_images_[0], mapped_buffer.get());
+  if (!SetRawImagePlanes(&raw_images_[0], mapped_buffer.get())) {
+    return {};
+  }
   prepared_buffers.push_back(mapped_buffer);
   for (size_t i = 1; i < encoders_.size(); ++i) {
     // Native buffers should implement optimized scaling and is the preferred
@@ -1539,7 +1546,9 @@
           << VideoFrameBufferTypeToString(mapped_buffer->type());
       return {};
     }
-    SetRawImagePlanes(&raw_images_[i], scaled_buffer.get());
+    if (!SetRawImagePlanes(&raw_images_[i], scaled_buffer.get())) {
+      return {};
+    }
     prepared_buffers.push_back(scaled_buffer);
   }
   return prepared_buffers;
